crawl-ref/source/dat/des/variable/float.des

```diff
@@ -4901,6 +4901,77 @@ MAP
 .cccccccccc
 ENDMAP
 
+NAME:   dreamdust_black_sun_cult
+TAGS:   transparent no_monster_gen
+ORIENT: float
+DEPTH:  D:6-10, Zot, !Zot:$
+WEIGHT: 3
+COLOUR: bm = red
+TILE:   m = dngn_transparent_wall_red
+FTILE:  -mA? = floor_volcanic
+KFEAT:  ! = .
+MARKER: ! = lua:fog_machine { cloud_type='flame', pow_min = 10, pow_max = 10, \
+                              delay = 10, size = 1, excl_rad = 0 }
+: if you.in_branch("D") then
+KMONS:  ? = sun demon
+KMONS:  - = nothing w:50 / demonspawn
+KITEM:  ? = manual of fire magic w:5 / \
+            randbook disc:fire disc2:necromancy spells:scorch \
+            slevels:8 numspells:4 title:The_Black_Sun, \
+            ring of protection from fire / ring of fire / \
+            staff of fire / scroll of immolation q:3
+: else
+KMONS:  ? = orb of fire
+KMONS:  - = nothing w:50 / black sun / \
+            draconian monk w:5 / draconian scorcher
+KITEM:  ? = manual of fire magic w:5 / \
+            randbook disc:fire disc2:necromancy spells:irradiate \
```

Comment:
Irradiate isn't consistent with this theme of "Necromancy and Fire" you've established; it's not even using the two schools of your custom spellbook. The name does sort of suggest Sun, but I'd recommend to sticking with spells that use one of the two established themes. Starburst is the most obvious spell to use here. It's also worth pointing out that you've actually gone against the theme of the game's existing notion of "Black Sun". The Black Sun of the classed demonspawn monster isn't one with a connection to Fire; it's a more figurative "Sun" that just radiates evil. The Black Mark ability that defines this demonspawn class does the following:

> Empowers the caster and any nearby allies to debilitate their foes on a
successful melee attack, allowing them to inflict draining, cause weakness, or
deplete magical power.

Although the more literal Fire theme you've established is consistent and well integrated into the vault, you've not really used draining, weakening, or anti-magic effects in loot or monsters that are central to the special black sun demonspawn outside of the one use of the black sun itself that only places in some locations. It's possible to use something like orange demon to apply weak to the player or have draining or anti-magic weapons at monster equipment or loot, for example.